pennylane_lightning/lightning_gpu/_measurements.py

```diff
@@ -138,6 +145,64 @@ def _process_single_shot(samples):
             tuple(zip(*processed_samples)) if shots.has_partitioned_shots else processed_samples[0]
         )
 
+    def expval(self, measurementprocess: MeasurementProcess):
+        """Expectation value of the supplied observable contained in the MeasurementProcess.
+
+        Args:
+            measurementprocess (StateMeasurement): measurement to apply to the state
+
+        Returns:
+            Expectation value of the observable
+        """
+
+        if isinstance(measurementprocess.obs, qml.SparseHamiltonian):
```

Comment:
What would you think of using `not cond` to avoid op nesting here?
```suggestion
        if not isinstance(measurementprocess.obs, qml.SparseHamiltonian):
            raise ValueError(...)
```

Reply:
I think It would not be possible to do that. This function returns for specific cases of measurements and all the `raises` are computed before.